### PR TITLE
Ec2 vpc cidr block

### PR DESCRIPTION
**Pull request by soumyo13**
### Description

With the VPC resource, now VPC CIDR Block Resource is also attached.

### Issues Resolved
https://github.com/inspec/inspec-aws/issues/314

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

**Comment by sathish-progress** [comment]
We would need matchers for single resource !!

**Comment by soumyo13** [comment]
> We would need matchers for single resource !!

Why we need this?

**Comment by sathish-progress** [comment]
> > We would need matchers for single resource !!
> 
> Why we need this?

Without this the users won't be able to easily check wether the CIDR blocks are associated!

**Comment by sathish-progress** [comment]
> > > We would need matchers for single resource !!
> > 
> > 
> > Why we need this?
> 
> Without this the users won't be able to easily check wether the CIDR blocks are associated!

example to build custom have_* matchers https://github.com/inspec/inspec-aws/blob/master/libraries/aws_s3_bucket.rb#L61

**Comment by soumyo13** [comment]
> > > > We would need matchers for single resource !!
> > > 
> > > 
> > > Why we need this?
> > 
> > 
> > Without this the users won't be able to easily check wether the CIDR blocks are associated!
> 
> example to build custom have_* matchers https://github.com/inspec/inspec-aws/blob/master/libraries/aws_s3_bucket.rb#L61

Added.

File: libraries/aws_vpcs.rb [hunk 1]
```diff
@@ -43,7 +45,13 @@ def fetch_data
...
                    is_default: vpc[:is_default],
                    instance_tenancy: vpc[:instance_tenancy],
-                   tags: map_tags(vpc[:tags]) }]
+                   tags: map_tags(vpc[:tags]),
+                   cidr_block_association_set: {
+                     cidr_block_association_set_association_ids: vpc[:cidr_block_association_set].first[:association_id],
+                     cidr_block_association_set_statuses: vpc[:cidr_block_association_set].first[:cidr_block_state][:state],
+                     cidr_block_association_set_status_messages: vpc[:cidr_block_association_set].first[:cidr_block_state][:status_message],
+                   },
+                   ipv_6_cidr_block_association_set: {} }]
```
Comment (sathish-progress) [inline libraries/aws_vpcs.rb:54]: should this be empty?

File: libraries/aws_vpc.rb [hunk 2]
```diff
@@ -49,4 +49,24 @@ def available?
...
+
+  def cidr_block_association_set
+    return [] unless vpc
+
+    vpc.cidr_block_association_set
+  end
+
+  def ipv_6_cidr_block_associated?
+    return true unless exists?
+    @vpc[:ipv_6_cidr_block_association_set] != nil
```
Comment (sathish-progress) [inline libraries/aws_vpc.rb:65]: `.nil?` is the preferred approach

File: libraries/aws_vpc.rb [hunk 3]
```diff
@@ -49,4 +49,24 @@ def available?
...
+  end
+
+  def vpc
+    return unless @display_name
```
Comment (sathish-progress) [inline libraries/aws_vpc.rb:69]: shouldn't this be `return unless exists?`

File: libraries/aws_vpcs.rb (hunk 1 above)
Comment (soumyo13) [reply libraries/aws_vpcs.rb:54]: All the functionalities of the ipv_6_cidr_block_association_set are already available in cidr_block_association_set. So if the customer wants it then we can give it.

Comment (sathish-progress) [reply libraries/aws_vpcs.rb:54]: then why duplicate the key please remove it ! also remove it from [here](https://github.com/inspec/inspec-aws/pull/336/files?file-filters%5B%5D=.rb#diff-808fa2d7d990d30a91c8a9c550c9e2c275c9e908e59bee2a161362480c0be6acR25)

File: libraries/aws_auto_scaling_group.rb [hunk 4]
```diff
@@ -34,11 +34,15 @@ def initialize(opts = {})
       @tags = []
 
       auto_scaling_group[:tags].map { |tag|
-        @tags.push({ 'resource_id':         tag.resource_id,
```
Comment (sathish-progress) [inline libraries/aws_auto_scaling_group.rb:37]: don't commit this file along with this PR.. I guess we already merged this change

File: libraries/aws_auto_scaling_groups.rb [hunk 5]
```diff
@@ -45,11 +45,15 @@ def fetch_data
                          health_check_type:         group[:health_check_type] }]
 
         group[:tags].map { |tag|
-          config_rows.push({ 'resource_id':          tag.resource_id,
```
Comment (sathish-progress) [inline libraries/aws_auto_scaling_groups.rb:48]: same [comment](https://github.com/inspec/inspec-aws/pull/336/files?file-filters%5B%5D=.rb#r637736611) here

**Review by sathish-progress — DISMISSED** [review]
Please review suggestions

File: libraries/aws_vpc.rb (hunk 3 above)
Comment (soumyo13) [reply libraries/aws_vpc.rb:69]: done

File: libraries/aws_vpc.rb (hunk 2 above)
Comment (soumyo13) [reply libraries/aws_vpc.rb:65]: done.

File: libraries/aws_vpcs.rb (hunk 1 above)
Comment (soumyo13) [reply libraries/aws_vpcs.rb:54]: as discussed we kept both the attributes as both attributes differs.

File: libraries/aws_vpc.rb (hunk 3 above)
Comment (sathish-progress) [reply libraries/aws_vpc.rb:69]: not done yet

File: docs/resources/aws_vpc.md [hunk 6]
```diff
@@ -51,34 +51,60 @@ See also the [AWS documentation on VPCs](https://docs.aws.amazon.com/vpc/latest/
...
+### Test if the IPV6 CIDR Block is associated to a named VPC
+
+    describe aws_vpc do
+      it { should have_ipv6_cidr_block_associated?(aws_ipv_6_cidr_block_association_set_ipv_6_cidr_block) }
+    end
+
+### Test if the CIDR Block is associated to a named VPC
+
+    describe aws_vpc do
+      it { should have_cidr_block_associated?(aws_cidr_block) }
```
[the remark this answers is not in the document]
Comment (sathish-progress) [reply docs/resources/aws_vpc.md:105]: yeah actually ..

**Review by Nirbhay1997 — APPROVED** [review]

**Review by soumyo13 — COMMENTED** [review]
Need to get the review from Satish.

**Pull request merged**